### PR TITLE
Implement `FlattenSubKeysd`

### LINKED ISSUE #5553: Promote Child Items to Top Level
The MONAO dict-based transforms acts on the top level keys, for instance `data={"image": ..., "pred": ...}`; however, when the output is nested (like in the hovernet model), the actual data will lie one level below like `data={"image": ..., "pred": {"a":..., "b":...}}`.

Currently, if user want to apply transform on "a" or "b", it can use `Lambdad` and apply that transfrom on `data["pred"]` instead and use "a" and "b" as keys. However, if the user want to apply a transform on "image" and "a" for instance, I didn't find any transform that can enable it.

Here a transform that can promote these children items ("a" and "b") to the same level of "image" can solve the problem.

### PR DESCRIPTION
**Pull request by drbeh**
Fixes #5553 

### Description

Promote specified children of specified items to the level of item.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by drbeh** [comment]
I am adding the unittests but you can find the use case [here](https://github.com/Project-MONAI/tutorials/blob/4d6bd79196efec96a37d00b5a668109d6b58e329/pathology/hovernet/infer_roi.py#L86-L90) o https://github.com/Project-MONAI/tutorials/pull/1055

**Comment by drbeh** [comment]
@wyli @Nic-Ma could you please help to review this PR as the progress of testing hovernet integration with MONAI label relies on this?

**Review by myron — COMMENTED** [review]
Hi @drbeh , it seems good to me, but the name PromoteChildItemsd() is not very intuitive.  You can consider calling it 
MergeSubKeys(sub_keys=...) or  FlattenSubKeys(sub_keys=..).  And if sub_keys is not provided, merge all of them. 

but this is minor, should not a be stopper for this issue..

**Comment by wyli** [comment]
agree with @myron, I think `PromoteChildItemsd` is not very intuitive. also, how to deal with duplicated sub-keys such as `{"a": {"a": 1, "b": 2}}`?

**Comment by drbeh** [comment]
@myron @wyli I have change the name, added the support to only specify the key, and handle situations where the same key exists at the top level.

File: monai/transforms/utility/dictionary.py [hunk 1]
```diff
@@ -779,6 +782,55 @@ def __call__(self, data):
...
 
+class FlattenSubKeysd(MapTransform):
+    """
+    If an item is dictionary, it flatten the item by moving the sub-items (defined by sub-keys) to the top level.
+    {"pred": {"a": ..., "b", ... }} --> {"a": ..., "b", ... }
+
+    Args:
+        keys: keys of the corresponding items to be flatten
+        sub_keys: the sub-keys of items to be flatten. If not provided all the sub-keys are flattened.
+        delete_keys: wether to delete the key of the items that their sub-keys are flattened. Default to True.
```
Comment (wyli) [inline monai/transforms/utility/dictionary.py:793]:
```suggestion
        delete_keys: whether to delete the key of the items that their sub-keys are flattened. Default to True.
```

File: monai/transforms/utility/dictionary.py [hunk 2]
```diff
@@ -779,6 +782,55 @@ def __call__(self, data):
...
+        d = dict(data)
+        for key in self.key_iterator(d):
+            # set the sub-keys for the specified key
+            sub_keys = d[key].keys() if self.sub_keys is None else self.sub_keys
+
+            # move all the sub-keys to the top level
+            for sk in sub_keys:
+                # set the top-level key for the sub-key
+                sk_top = f"{self.prefix}_{sk}" if self.prefix else sk
+                if sk_top in d.keys():
```
Comment (wyli) [inline monai/transforms/utility/dictionary.py:822]:
```suggestion
                if sk_top in d:
```

**Review by wyli — APPROVED** [review]
looks good to me, please fix the typos I'll merge this

**Pull request merged**